### ISSUE TITLE
Unsafe autocorrection for Lint/SafeNavigationChain

### ISSUE THREAD

**Issue by Huy-Ngo** [rubocop/rubocop#14985]
Autocorrection for Lint/SafeNavigationChain can sometimes lead to syntax error.

--------

## Steps to reproduce the problem

I encounter this error when I run rubocop on some code which includes ternary operator on an ActiveModel object:

```rb
user = User.find_by(uuid: params[:uuid])

some_number = user&.age ? user&.age - 1 : 0
```

## Expected behavior

It should either ignore this, or detect but shouldn't autocorrect it

## Actual behavior

The autocorrection leads to syntax error.

Command output:

```sh
$ rubocop dummy.rb --autocorrect
Inspecting 1 file
F

Offenses:

dummy.rb:3:1: W: [Corrected] Lint/UselessAssignment: Useless assignment to variable - some_number.
some_number = user&.age ? user&.age - 1 : 0
^^^^^^^^^^^
dummy.rb:3:36: W: [Corrected] Lint/SafeNavigationChain: Do not chain ordinary method call after safe navigation operator.
some_number = user&.age ? user&.age - 1 : 0
                                   ^^^^
dummy.rb:3:40: F: Lint/Syntax: expected a : after the true expression of a ternary operator
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
some_number = user&.age ? user&.age&. - 1 : 0

dummy.rb:3:43: F: Lint/Syntax: unexpected ':', expecting end-of-input
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
some_number = user&.age ? user&.age&. - 1 : 0
                                          ^

1 file inspected, 4 offenses detected, 2 offenses corrected
```

Resulted file:

```rb
user = User.find_by(uuid: params[:uuid])

some_number = user&.age ? user&.age&. - 1 : 0
```

## RuboCop version

1.82.1 (using Parser 3.3.10.0, Prism 1.7.0, rubocop-ast 1.49.0, analyzing as Ruby 3.4, running on ruby 3.4.2) [x86_64-linux]
  - rubocop-rails 2.34.2
  - rubocop-rspec 3.8.0
  - rubocop-performance 1.26.1
